CI 3.0-dev compatibility update for Router.php

hArpanet opened the pull request:
CI 3.0...

Requires controller names to begin with uppercase;
Requires URI segments to start at 1 not 0;

Both changes make a check against CI version for backwards compatibility. 

Note that the upper-casing of the controller names is done using an anonymous function, which requires PHP version 5.3.0

Patroklo committed:
Change for CI 3.0

This simple change let's using this library into Codeigniter 3.0

hArpanet committed:
CI 3.0-dev compatibility update for Router.php

CI 3.0...
requires controller named to begin with uppercase;
requires URI segments to start at 1 not 0;

hArpanet commented:
@Patroklo Just pinging you in case you are not automatically notified of my pull request, which you might like to include in your branch too.

jenssegers commented:
Is CI 3 the current stable release? I haven't used CI for 2 years or something :)

hArpanet committed:
Detect default_controller in Module

Allows the default_controller named in config/routes.php to be located
within a module of the same name (eg.
modules/welcome/controllers/Welcome.php)

hArpanet commented:
Hmm, depends on how you interpret 'stable release' :-)

I did a full review of the current changes in the 3.0-dev branch against the current 'stable release' (v2 branch) and following some comments from others I can see no reason why new installs of CI would not use the current 3.0-dev branch. I'm in the process of using it for production here.

hArpanet commented:
I have a couple more pull requests to send you as well, but I don't want to include them in this one (unless you can pick and choose commits to apply, which I don't know is possible).

jenssegers commented:
I tagged a v1.0.0 release for CI2 and created a 2.x branch so that the old code is still available.
